ProcessLib/Parameter/SpatialPosition.h

```diff
@@ -0,0 +1,93 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef PROCESSLIB_SPATIALPOSITION_H
+#define PROCESSLIB_SPATIALPOSITION_H
+
+#include <boost/optional.hpp>
+#include "MathLib/TemplatePoint.h"
+
+namespace ProcessLib
+{
+//! Represents a position in space which can be either one of
+//! a node, an element, an integration point or a cartesian coordinates triple.
+//!
+//! The setters of this class make sure that only compatible information can be
+//! stored at the same time; e.g., it is not possible to specify an element ID
+//! and a node ID at the same time (the setAll() method being an exception to
+//! that rule).
+class SpatialPosition
```

Comment:
Interesting observation. `SpatialPosition` is like a combination of a `MeshLib::Location` and a point. Let's talk on monday.